src/scheduler/createNotebookScheduler.tsx

```diff
@@ -0,0 +1,764 @@
+import { MuiChipsInput } from 'mui-chips-input';
+import { IThemeManager } from '@jupyterlab/apputils';
+import { JupyterLab } from '@jupyterlab/application';
+import LabelProperties from '../jobs/labelProperties';
+import { v4 as uuidv4 } from 'uuid';
+import { Cron } from 'react-js-cron';
+import 'react-js-cron/dist/styles.css';
+import { KernelSpecAPI } from '@jupyterlab/services';
+import tzdata from 'tzdata';
+import { SchedulerService } from './schedulerServices';
+import NotebookJobComponent from './notebookJobs';
+import LeftArrowIcon from '../../style/icons/left_arrow_icon.svg';
+import { LabIcon } from '@jupyterlab/ui-components';
+import errorIcon from '../../style/icons/error_icon.svg';
+import { Button } from '@mui/material';
+import { scheduleMode } from '../utils/const';
+
+interface IDagList {
+  jobid:  string;
+  notebookname:  string;
+  schedule:  string;
+  scheduleInterval:string;
+}
+
+const iconLeftArrow = new LabIcon({
+  name: 'launcher:left-arrow-icon',
+  svgstr: LeftArrowIcon
+});
+
+const iconError = new LabIcon({
+  name: 'launcher:error-icon',
+  svgstr: errorIcon
+});
+
+const CreateNotebookScheduler = ({
+  themeManager,
+  app,
+  context
+}: {
+  themeManager: IThemeManager;
+  app: JupyterLab;
+  context: any;
+}): JSX.Element => {
+  const [jobNameSelected, setJobNameSelected] = useState('');
+  const [inputFileSelected, setInputFileSelected] = useState('');
+  const [composerList, setComposerList] = useState<string[]>([]);
+  const [composerSelected, setComposerSelected] = useState('');
+
+  const [parameterDetail, setParameterDetail] = useState(['']);
+  const [parameterDetailUpdated, setParameterDetailUpdated] = useState(['']);
+  const [keyValidation, setKeyValidation] = useState(-1);
+  const [valueValidation, setValueValidation] = useState(-1);
+  const [duplicateKeyError, setDuplicateKeyError] = useState(-1);
+
+  const [selectedMode, setSelectedMode] = useState('cluster');
+  const [clusterList, setClusterList] = useState<string[]>([]);
+  const [serverlessList, setServerlessList] = useState<string[]>([]);
+  const [serverlessDataList, setServerlessDataList] = useState<string[]>([]);
+  const [clusterSelected, setClusterSelected] = useState('');
+  const [serverlessSelected, setServerlessSelected] = useState('');
+  const [serverlessDataSelected, setServerlessDataSelected] = useState({});
+  const [stopCluster, setStopCluster] = useState(false);
+
+  const [retryCount, setRetryCount] = useState<number | undefined>(2);
+  const [retryDelay, setRetryDelay] = useState<number | undefined>(5);
+  const [emailOnFailure, setEmailOnFailure] = useState(false);
+  const [emailOnRetry, setEmailonRetry] = useState(false);
+  const [emailOnSuccess, setEmailOnSuccess] = useState(false);
+  const [emailList, setEmailList] = useState<string[]>([]);
+
+  const [scheduleMode, setScheduleMode] = useState<scheduleMode>('runNow');
+  const [scheduleValue, setScheduleValue] = useState('30 17 * * 1-5');
```

Comment:
We should avoid using magic numbers here. Put this into a constant with a description.

Reply:
Resolved - moved to constant and description added